spring-boot-test-autoconfigure/src/test/resources/application.properties

```diff
@@ -0,0 +1,2 @@
+# Use random port for embedded Mongod.
+spring.data.mongodb.port = 0
```

Comment:
That change isn't related to `@DataMongoTest` but by the port used flapdoodle. Without specifying 0 for random ports, it used the same port for all tests when on the test class path for `spring-boot-test-autoconfigure` which collided with other tests, therefore I set this property. But it's only for the tests of `spring-boot-test-autoconfigure`